### PR TITLE
Add EIP: Network Upgrade Inclusion Stages

### PR DESCRIPTION
**Pull request by timbeiko**
Meta EIP to define `Proposed for Inclusion`, `Considered for Inclusion`, `Schedule for Inclusion` and `Included`. When merged, this EIP should replace [this](https://github.com/ethereum/execution-specs/blob/master/network-upgrades/README.md).

### CODE REVIEW

File: EIPS/eip-cfi.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+---
```
Comment (abcoathup) [inline EIPS/eip-cfi.md:1]:
```suggestion
---
eip: 7723
```
Assigning next sequential EIP/ERC/RIP number.

Please also update the filename.

File: EIPS/eip-cfi.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+title: Network Upgrade Inclusion Stages
+description: Overview of the various stages Core EIPs go through before their activation in network upgrades.
+author: Tim Beiko (@timbeiko)
+discussions-to: https://ethereum-magicians.org/t/eip-xxxx-network-upgrade-inclusion-stages/20281
```
Comment (abcoathup) [inline EIPS/eip-cfi.md:5]:
```suggestion
discussions-to: https://ethereum-magicians.org/t/eip-7723-network-upgrade-inclusion-stages/20281
```

**Comment by timbeiko** [comment]
Thanks @abcoathup, updated with the number and based on your feedback on EthMag

**Comment by gorondan** [comment]
> Meta EIP to define `Proposed for Inclusion`, `Considered for Inclusion`, `Schedule for Inclusion` and `Included`. When merged, this EIP should replace [this](https://github.com/ethereum/execution-specs/blob/master/network-upgrades/README.md).

These changes will bring more clarity in tracking the status of an EIP during the network upgrade process :+1:

File: EIPS/eip-7723.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+Before the Upgrade Meta EIP is moved to `Final`, the `Scheduled for Inclusion` stage **MUST** be renamed to `Included` and contain only EIPs that were activated in the upgrade. 
+
+### Proposed for Inclusion
+
+To formally propose the inclusion of a Core EIP in a network upgrade, someone **MUST** open a pull request against the Upgrade Meta EIP to add the EIP in the `Proposed for Inclusion` section. Reasonable pull requests **SHOULD** be merged in a timely fashion by the Upgrade Meta EIP author.  
+
+At this stage, implementation teams **SHOULD** review the EIP in the context of including it in the next upgrade. 
+
+Note that EIPs must be `Proposed for Inclusion` for each network upgrade. In other words, proposals do not "carry over" to the next upgrade if an EIP is not included in the one it was first proposed for.
```
Comment (non-fungible-nelson) [inline EIPS/eip-7723.md:42]: Is it out of scope for this proposal to categorize a rolling way to track community interest in certain enhancements for the next upgrade? Cataloguing this on a continuous basis might help reduce the perceived scope creep and crunch around fork scheduling time, which is infrequent enough that it causes frustration with the process.

We really only do this for fork drivers like Verkle.

Comment (non-fungible-nelson) [reply EIPS/eip-7723.md:42]: Maybe could be solved as easily as keeping an N+1 Meta-EIP model for forks on the repo

Comment (abcoathup) [reply EIPS/eip-7723.md:42]: > Maybe could be solved as easily as keeping an N+1 Meta-EIP model for forks on the repo

@non-fungible-nelson Meta EIP for Osaka already exists
https://eips.ethereum.org/EIPS/eip-7607

Comment (timbeiko) [reply EIPS/eip-7723.md:42]: I'm not sure I'd want this as part of this EIP, but a simple rule could be:

Current fork: proposals can move up to SFI
Next fork: proposals can move up to CFI
Fork +2: things can be PFI'd

That said, my sense is most things in Fork+2 would end up stale by the time we actually get to it, so I prefer having only two at the time.

File: EIPS/eip-7723.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+### Scheduled for Inclusion 
+
+When client developers decide to work on an EIP as part of a network upgrade, the EIP **SHOULD** move to the `Scheduled for Inclusion` stage, and the Upgrade Meta EIP **MUST** be updated to reflect this. 
+
+`Scheduled for Inclusion` signals that implementation and testing work are underway, and that, assuming no issues arise as part of this process, the EIP **SHOULD** be included in the network upgrade. An EIP **MAY** be removed from `Scheduled for Inclusion` if client teams are generally against including the EIP in the network upgrade. 
+
+### Included
+
+Once a network upgrade has been activated, all EIPs that were part in the upgrade **MUST** be moved to `Included` in the Upgrade Meta EIP, and the `Proposed for Inclusion`, `Considered for Inclusion` and `Scheduled for Inclusion` lists **MUST** be removed from the Meta EIP.
```
Comment (dapplion) [inline EIPS/eip-7723.md:58]: I am not sure we need 3 states. Status `CFI` and `SFI` sound confusing. If an EIP is `CFI` should clients implement it already or not? If not, what's the point of this status? Also, historically clients have implemented EIPs at different times depending on their priorities, regardless of overall consensus.

I assume all EIPs in existence are meant to be included eventually, so all EIPs are automatically `Proposed for Inclusion`. Following the proposed work-flow of doing PRs against the meta-EIP, the existence of this PR already shows intent for the EIP to be scheduled.

Why not have a single list: `Scheduled for Inclusion`? Consider the following flow:

- EIP is drafted
- EIP champion opens PR to the meta-EIP (= shows intent to be scheduled)
- Client developers, community members, and affected stakeholders can signal positive intent on the PR / forum
- Once rough consensus forms for inclusion, the PR is merged. At this point the EIP is expected to be implemented by clients

Comment (timbeiko) [reply EIPS/eip-7723.md:58]: @dapplion I replied to this on [EthMag](https://ethereum-magicians.org/t/eip-7723-network-upgrade-inclusion-stages/20281/5?u=timbeiko)

File: EIPS/eip-7723.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+### Scheduled for Inclusion 
+
+When client teams decide to work on an EIP as part of a network upgrade, the EIP **SHOULD** move to the `Scheduled for Inclusion` stage, and the Upgrade Meta EIP **MUST** be updated to reflect this. 
+
+`Scheduled for Inclusion` signals that implementation and testing work are underway, and that, assuming no issues arise as part of this process, the EIP **SHOULD** be included in the network upgrade. An EIP **MAY** be removed from `Scheduled for Inclusion` if client teams are generally against including the EIP in the network upgrade. 
+
+### Included
+
+Once a network upgrade has been activated, all EIPs that were part in the upgrade **MUST** be moved to `Included` in the Upgrade Meta EIP, and the `Proposed for Inclusion`, `Considered for Inclusion` and `Scheduled for Inclusion` lists **MUST** be removed from the Meta EIP.
```
Comment (abcoathup) [inline EIPS/eip-7723.md:58]:
```suggestion
Once a network upgrade has been activated, all EIPs that were part of the upgrade **MUST** be moved to `Included` in the Upgrade Meta EIP, and the `Proposed for Inclusion`, `Considered for Inclusion` and `Scheduled for Inclusion` lists **MUST** be removed from the Meta EIP. 
```
Minor grammar fix

**Review by abcoathup — APPROVED** [review]
Other than one minor grammar issue, LGTM

**Review by SamWilsn — COMMENTED** [review]
Meta proposals can completely ignore `eipw`, so set the sections up in a way that makes the most sense for this document. Just keep abstract and copyright.

I'd use _italics_ instead of `backticks` for your keywords, since backticks are generally for inline code.

File: EIPS/eip-cfi.md (hunk 1 above)
Comment (Parasos) [reply EIPS/eip-cfi.md]: > ```suggestion
> ---
> eip: 7723
> ```
> Assigning next sequential EIP/ERC/RIP number.
> 
> Please also update the filename.
